pkg/addon-operator/operator.go

```diff
@@ -1398,10 +1444,23 @@ func (op *AddonOperator) HandleGlobalHookRun(t sh_task.Task, labels map[string]s
 		op.MetricStorage.HistogramObserve("{PREFIX}global_hook_run_seconds", d.Seconds(), metricLabels)
 	})()
 
-	logEntry.Infof("Global hook run")
+	isSynchronization := hm.BindingType == OnKubernetesEvent && hm.BindingContext[0].Type == types.TypeSynchronization
+	shouldRunHook := true
```

Comment:
Can we move this code to a method for this instead of copying it? And why do we add `shouldRunHook` by the way? :)

Reply:
1. I've created a method IsSynchronization.

2. `shouldRunHook` is needed as GlobalHookRun and ModuleHookRun tasks are now created for bindings with `executeHookOnSynchronization: false`. These tasks only enable emitting events from the binding's Monitor and do not run the hook.

Reply:
About the second part:
I was wondering about moving some code out of the `Handle*` methods. If it is hard or gives no benefits, let's leave things as is.